### ISSUE TITLE
Not seeing green download arrow

### ISSUE THREAD

**Issue by interstar** [esteinmann/chatgpt-convdown#29]
It's not appearing for me, Firefox on Ubuntu.

Working with ChatGPT4 in the paid account.

Has this plugin been broken by changes to Chat's HTML layout?

Is there another way to trigger a download?

**Comment by interstar** [comment]
Actually it seems to appear instantaneously when I reload page and then disappears again almost immediately

**Comment by esteinmann** [comment]
Thanks for reporting this issue. Is ChatGPT4 still on the same page? I don't have a paid account and haven't looked at ChatGPT4 yet. When I visit https://chat.openai.com/chat it still looks the same as before. I see: ChatGPT Mar 14 Version. Free Research Preview at the bottom. Same combo of browser and OS (well Kubuntu but that shouldn't make a difference)

Although I can confirm the behavior you specify in the second comment. It seems like that another ('soft') reload happens after the chat contents is loaded. When that happens the button does not seems to return. Will investigate this but can't say anything meaningful on ChatGPT4 related issues for now.

**Comment by esteinmann** [comment]
To answer another question, there currently is no other way to trigger a download. Might me good to have another option. Actually the first version of this extension used the context menu of the browser for triggering the download, maybe that wasn't so bad and something to bring back.

**Comment by esteinmann** [comment]
That shouldn't happen, the chat history list seems to slide under the bottom part of the nav area. The download button is added to the sticky part of the nav.

I wonder how you see the page when no chat is selected yet. For me it looks like this:
![afbeelding](https://user-images.githubusercontent.com/63729650/226735168-847e68fd-c760-4437-9a81-1259a3dd6b81.png)

Do you see the download button when no chat is selected yet as in my screenshot?

**Comment by interstar** [comment]
Yes. When there's no chat, it's like this. Even when GPT4 is selected in dropdown

![image](https://user-images.githubusercontent.com/90998/226972717-ff2aa7ef-b3fb-42e1-91a2-73adfd73ce65.png)

Actually, it seems to work with a new chat. (And the button is working fine)

![image](https://user-images.githubusercontent.com/90998/226974057-f39b4cfb-bb4b-47c3-83a6-40a9f839bbfd.png)

I can't test loading an old history because histories are currently unavailable. So I wonder if it's only old histories that have the problem. Or maybe there's been some tweak to the chat html UI.

**Comment by BirdCatcherGames** [comment]
I can trigger this bug consistently with long chat histories. I have a thread that was downloadable at ~60k words, but now that it's over 100k words the button no longer appears.

**Comment by esteinmann** [comment]
@BirdCatcherGames 
It's a bit late but I noticed the same when the chats take a while to load from the backend API. Experimented with catching the web requests to the backend and re-add the button when they finish loading. Put this in a separate branch as this requires adding the webRequest permission.

https://github.com/esteinmann/chatgpt-convdown/tree/experimental/web-request-listener
